modules/security-deploy-image-sources.adoc

```diff
@@ -0,0 +1,74 @@
+// Module included in the following assemblies:
+//
+// * security/container_security/security-deploy.adoc
+
+[id="security-deploy-image-sources_{context}"]
+= Controlling what image sources can be deployed
+
+It is important that the intended images are actually being deployed, that the
+images including the contained content
+are from trusted sources, and they have not been altered. Cryptographic signing
+provides this assurance. {product-title} enables cluster administrators to apply
+security policy that is broad or narrow, reflecting deployment environment and
+security requirements. Two parameters define this policy:
+
+- one or more registries (with optional project namespace)
+- trust type (accept, reject, or require public key(s))
+
+You can use these policy parameters to allow, deny, or require a trust
+relationship for entire registries, parts of registries, or individual
+images. Using trusted public keys, you can ensure that the source is
+cryptographically verified.
+The policy rules apply to nodes. Policy may be
+applied uniformly across all nodes or targeted for different node workloads (for
+example, build, zone, or environment).
+
+.Example image signature policy file
+[source,yaml]
+----
+{
+    "default": [{"type": "reject"}],
+    "transports": {
+        "docker": {
+            "access.redhat.com": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyPath": "/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release"
+                }
+            ]
+        },
+        "atomic": {
+            "172.30.1.1:5000/openshift": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyPath": "/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release"
+                }
+            ],
+            "172.30.1.1:5000/production": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyPath": "/etc/pki/example.com/pubkey"
+                }
+            ],
+            "172.30.1.1:5000": [{"type": "insecureAcceptAnything"}]
+        }
+    }
+}
+----
+
+The policy can be saved onto a node as *_/etc/containers/policy.json_*. This
+example enforces the following rules:
+
+. Require images from the Red Hat Registry (`registry.access.redhat.com`) to be
+signed by the Red Hat public key.
+. Require images from your OpenShift Container Registry in the *openshift*
+namespace to be signed by the Red Hat public key.
+. Require images from your OpenShift Container Registry in the *production*
+namespace to be signed by the public key for `example.com`.
+. Reject all other registries not specified by the global `default` definition.
+
+For more details on image signature policy, see
+link:https://github.com/containers/image/blob/master/docs/containers-policy.json.5.md[syntax for the signature verification file] source code documentation.
```

Comment:
So we have a rule with the 4.x docs that we're not supposed to link to GitHub, unless we've run the proposed links by Eric Rich. Basically to make sure that we're not pointing to things that really aren't actually supported.

So for all the github.com links you have in this PR, you'll either have to remove them if they're really not necessary, see whether there is elsewhere in the doc set that might have this info that you can point to, or run them by Eric for approval.

Reply:
I don't trust that this link is up to date anyway. I'll cut it.